app/screens/table_image/table_image.js

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+    ActivityIndicator,
+    Image,
+    ScrollView,
+    StyleSheet,
+    View
+} from 'react-native';
+
+export default class TableImage extends React.PureComponent {
+    static propTypes = {
+        deviceWidth: PropTypes.number.isRequired,
+        imageSource: PropTypes.string.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            width: -1,
+            height: -1
+        };
+    }
+
+    componentWillMount() {
+        this.getImageSize(this.props.imageSource);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.imageSource !== nextProps.imageSource) {
+            this.setState({
+                width: -1,
+                height: -1
+            });
+
+            this.getImageSize(nextProps.imageSource);
+        }
+    }
+
+    getImageSize = (imageSource) => {
+        Image.getSize(imageSource, (width, height) => {
+            this.setState({
+                width,
+                height
+            });
+        });
+    }
+
+    render() {
+        let {width, height} = this.state;
+
+        if (width === -1 || height === -1) {
+            return (
+                <View style={style.loadingContainer}>
+                    <ActivityIndicator
+                        animating={true}
+                        size='large'
+                    />
+                </View>
+            );
+        }
+
+        if (width > this.props.deviceWidth) {
+            height = (height / width) * this.props.deviceWidth;
+            width = this.props.deviceWidth;
+        }
+
+        return (
+            <ScrollView
+                style={style.scrollContainer}
+                contentContainerStyle={style.container}
+            >
+                <Image
```

Comment:
Maybe we can add [Zoom](https://facebook.github.io/react-native/docs/scrollview.html#zoomscale) and look for an alternative for Android

Reply:
How does that work? Do I just add a `minimumZoomScale` and `maximumZoomScale`?

I wasn't wanting to spend too much time adding functionality here yet since it's for a particularly narrow behaviour

Reply:
I believe @jarredwitt uses it in the image preview, perhaps take a look at that piece of the code